src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests4.cs

```diff
@@ -0,0 +1,663 @@
+    {
+        object t = new C();
+        Console.WriteLine(t is (3, 4, 5));
+    }
+}
+";
+            // Use a version of the platform APIs that lack ITuple
+            var compilation = CreateCompilationWithMscorlib40AndSystemCore(source, options: TestOptions.ReleaseExe);
+            compilation.VerifyDiagnostics(
+                // (17,32): error CS1061: 'object' does not contain a definition for 'Deconstruct' and no accessible extension method 'Deconstruct' accepting a first argument of type 'object' could be found (are you missing a using directive or an assembly reference?)
+                //         Console.WriteLine(t is (3, 4, 5));
+                Diagnostic(ErrorCode.ERR_NoSuchMemberOrExtension, "(3, 4, 5)").WithArguments("object", "Deconstruct").WithLocation(17, 32),
+                // (17,32): error CS8129: No suitable Deconstruct instance or extension method was found for type 'object', with 3 out parameters and a void return type.
+                //         Console.WriteLine(t is (3, 4, 5));
+                Diagnostic(ErrorCode.ERR_MissingDeconstruct, "(3, 4, 5)").WithArguments("object", "3").WithLocation(17, 32)
+                );
+        }
+
+        [Fact]
+        public void ITupleLacksIndexer()
+        {
+            // - should give an error when ITuple is missing required member (indexer)
+            var source =
+@"using System;
+namespace System.Runtime.CompilerServices
+{
+    public interface ITuple
+    {
+        int Length { get; }
+        // object this[int index] { get; }
+    }
+}
+public class C : System.Runtime.CompilerServices.ITuple
+{
+    int System.Runtime.CompilerServices.ITuple.Length => 3;
+    // object System.Runtime.CompilerServices.ITuple.this[int i] => i + 3;
+    public static void Main()
+    {
+        object t = new C();
+        Console.WriteLine(t is (3, 4, 5));
+    }
+}
+";
+            // Use a version of the platform APIs that lack ITuple
+            var compilation = CreateCompilationWithMscorlib40AndSystemCore(source, options: TestOptions.ReleaseExe);
+            compilation.VerifyDiagnostics(
+                // (17,32): error CS1061: 'object' does not contain a definition for 'Deconstruct' and no accessible extension method 'Deconstruct' accepting a first argument of type 'object' could be found (are you missing a using directive or an assembly reference?)
+                //         Console.WriteLine(t is (3, 4, 5));
+                Diagnostic(ErrorCode.ERR_NoSuchMemberOrExtension, "(3, 4, 5)").WithArguments("object", "Deconstruct").WithLocation(17, 32),
+                // (17,32): error CS8129: No suitable Deconstruct instance or extension method was found for type 'object', with 3 out parameters and a void return type.
+                //         Console.WriteLine(t is (3, 4, 5));
+                Diagnostic(ErrorCode.ERR_MissingDeconstruct, "(3, 4, 5)").WithArguments("object", "3").WithLocation(17, 32)
+                );
+        }
+    }
+}
```

Comment:
Could you add a test with obsolete ITuple interface?

Reply:
Sure, but I wouldn't expect an error. The error would appear on the API that declares that it implements the interface.

---
In reply to: [206290578](https://github.com/dotnet/roslyn/pull/28859#discussion_r206290578) [](ancestors = 206290578)